tools/build_cpp.py

```diff
@@ -0,0 +1,129 @@
+    return subprocess.check_call(cmd, stderr=subprocess.STDOUT, shell=shell)
+
+
+def main(prefix='install', build_dir='build', source_dir='.'):
+    """
+    Platform-independent function to run cmake, build, install and C++ tests.
+    """
+
+    # Get the platform name: 'linux', 'darwin' (osx), or 'win32'.
+    platform = sys.platform
+
+    # Set up absolute directory paths
+    source_dir = os.path.abspath(source_dir)
+    prefix = os.path.abspath(prefix)
+    build_dir = os.path.abspath(build_dir)
+
+    # Default options
+    shell = False
+    parallel_flag = '-j{}'.format(multiprocessing.cpu_count())
+    build_config = ''
+
+    # Some flags use a syntax with a space separator instead of '='
+    use_space = ['-G', '-A']
+
+    # Default cmake flags
+    cmake_flags = {
+        '-G': 'Ninja',
+        '-DPYTHON_EXECUTABLE': shutil.which("python"),
+        '-DCMAKE_INSTALL_PREFIX': prefix,
+        '-DWITH_CTEST': 'OFF',
+        '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION': 'ON'
+    }
+
+    if platform == 'darwin':
+        cmake_flags.update({'-DCMAKE_INTERPROCEDURAL_OPTIMIZATION': 'OFF'})
+        osxversion = os.environ.get('OSX_VERSION')
+        if osxversion is not None:
+            cmake_flags.update({
+                '-DCMAKE_OSX_DEPLOYMENT_TARGET':
+                osxversion,
+                '-DCMAKE_OSX_SYSROOT':
+                os.path.join('/Applications', 'Xcode.app', 'Contents',
+                             'Developer', 'Platforms', 'MacOSX.platform',
+                             'Developer', 'SDKs',
+                             'MacOSX{}.sdk'.format(osxversion))
+            })
+
+    if platform == 'win32':
+        cmake_flags.update({'-G': 'Visual Studio 16 2019', '-A': 'x64'})
+        shell = True
+        build_config = 'Release'
+
+    # Additional flags for --build commands
+    build_flags = [parallel_flag]
+    if len(build_config) > 0:
+        build_flags += ['--config', build_config]
+
+    # Parse cmake flags
+    flags_list = []
+    for key, value in cmake_flags.items():
+        if key in use_space:
+            flags_list += [key, value]
+        else:
+            flags_list.append('{}={}'.format(key, value))
+
+    if not os.path.exists(build_dir):
+        os.makedirs(build_dir)
+    os.chdir(build_dir)
+
+    # Run cmake
+    run_command(['cmake'] + flags_list + [source_dir], shell=shell)
+
+    # Show cmake settings
+    run_command(['cmake', '-B', '.', '-S', source_dir, '-LA'], shell=shell)
+
+    # Compile benchmarks
+    run_command(['cmake', '--build', '.', '--target', 'all-benchmarks'] +
+                build_flags,
+                shell=shell)
+
+    # Compile C++ tests
+    run_command(['cmake', '--build', '.', '--target', 'all-tests'] +
+                build_flags,
+                shell=shell)
+
+    # Compile Python library
+    run_command(['cmake', '--build', '.', '--target', 'install'] + build_flags,
+                shell=shell)
```

Comment:
```suggestion
    for target in ['all-benchmarks', 'all-tests', 'install']:
        run_command(['cmake', '--build', '.', '--target', target] + build_flags, shell=shell)
```
and similar below for the test runs?